src/test/java/ladder/domain/ns/NsLinesTest.java

```diff
@@ -0,0 +1,85 @@
+package ladder.domain.ns;
+
+import ladder.domain.engine.*;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static ladder.domain.engine.PlayerNameTest.*;
+import static ladder.domain.engine.PlayersTest.POBI_HONUX_CRONG_JK;
+import static ladder.domain.ns.NsLadderTest.VERTICAL_LADDER_SIZE;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+public class NsLinesTest {
+
+    @Test
+    @DisplayName("라인목록을 생성한다")
+    void create() {
+        Players players = new Players(POBI_HONUX_CRONG_JK);
+        Lines actual = new NsLines(players, VERTICAL_LADDER_SIZE, () -> false);
+        Line falseLine = new NsLine(players.size(), () -> false);
+        Lines expected = new NsLines(List.of(falseLine, falseLine, falseLine, falseLine, falseLine));
+
+        assertThat(actual).isEqualTo(expected);
+    }
+
+    @Test
+    @DisplayName("라인목록을 조회한다")
+    void getLines() {
+        Players players = new Players(POBI_HONUX_CRONG_JK);
+        Line falseLine = new NsLine(players.size(), () -> false);
+        ArrayList<Line> lineList = new ArrayList<>(List.of(falseLine, falseLine, falseLine, falseLine, falseLine));
+        Lines lines = new NsLines(lineList);
+        List<Line> actual = lines.getLines();
+
+        assertThat(actual).isEqualTo(lineList);
+        assertThatThrownBy(() -> {
+            actual.add(new NsLine(players.size(), () -> false));
+        }).isInstanceOf(UnsupportedOperationException.class);
+    }
+
+    @Test
+    @DisplayName("플레이어 모두 마지막까지 이동")
+    void move() {
+        //    |-----|     |-----| pobi  honux crong jk    0
+        //    |     |-----|     | honux pobi  jk    crong 1
+        //    |-----|     |     | honux jk    pobi  crong 2
+        //    |     |-----|     | jk    honux pobi  crong 3
+        //    |-----|     |-----| jk    pobi  honux crong 4
+        //    results             pobi  jk    crong honux 5
+        Point p1_1 = Point.first(true),
+                p1_2 = p1_1.next(false),
+                p1_3 = p1_2.next(true),
+                p1_4 = p1_3.last();
+
+        Point p2_1 = Point.first(false),
+                p2_2 = p2_1.next(true),
+                p2_3 = p2_2.next(false),
+                p2_4 = p2_3.last();
+
+        Point p3_1 = Point.first(true),
+                p3_2 = p3_1.next(false),
+                p3_3 = p3_2.next(false),
+                p3_4 = p3_3.last();
+
+        Lines lines = new NsLines(List.of(
+                new NsLine(List.of(p1_1, p1_2, p1_3, p1_4)),
```

Comment:
```suggestion
                new NsLine(true, false, true),
```

위와 같이 NsLine을 생성할 수 있는 생성자가 있다면 NsLine을 테스트 fixture를 좀 더 수월하게 생성할 수 있지 않을까?

Reply:
생성자를 추가해서 해소할수 있는 문제였었네요😅

fixture 을 줄이는 고민을 외부에서만 생각했었는데,
객체 자체에 생성자를 추가하는 방향을 더 유연하게 고민해봐야 할것 같습니다.
코멘트 감사드립니다🙇